t/rpcapi.t

```diff
@@ -0,0 +1,85 @@
+use strict;
+use warnings;
+use 5.14.2;
+
+use Test::More tests => 2;
+use Test::Exception;
+use Test::NoWarnings;
+
+my $TIME = CORE::time();
+BEGIN {
+    *CORE::GLOBAL::time = sub { $TIME };
+}
+
+use JSON::PP;
+use File::ShareDir qw[dist_file];
+use File::Temp qw[tempdir];
+
+use Zonemaster::Engine;
+use Zonemaster::Backend::Config;
+use Zonemaster::Backend::RPCAPI;
+
+sub advance_time {
+    my ( $delta ) = @_;
+    $TIME += $delta;
+}
+
+my $db_backend = Zonemaster::Backend::Config->check_db( $ENV{TARGET} || 'SQLite' );
+diag "database: $db_backend";
+
+my $tempdir = tempdir( CLEANUP => 1 );
+my $config = Zonemaster::Backend::Config->parse( <<EOF );
+[DB]
+engine = $db_backend
+
+[MYSQL]
+host     = localhost
+user     = travis_zm
+password = travis_zonemaster
+database = travis_zonemaster
+
+[POSTGRESQL]
+host     = localhost
+user     = travis_zonemaster
+password = travis_zonemaster
+database = travis_zonemaster
+
+[SQLITE]
+database_file = $tempdir/zonemaster.sqlite
+
+[ZONEMASTER]
+age_reuse_previous_test = 10
+EOF
+
+subtest 'Everything but Test::NoWarnings' => sub {
+
+    my $rpcapi = Zonemaster::Backend::RPCAPI->new(
+        {
+            dbtype => $db_backend,
+            config => $config,
+        }
+    );
+
+    if ( $db_backend eq 'SQLite' ) {
+        $rpcapi->{db}->create_db()
+          or BAIL_OUT( "$db_backend database could not be created" );
+    }
+
+    subtest 'start_domain_test' => sub {
+        my $testid1 = $rpcapi->start_domain_test( { domain => "zone1.rpcapi.example" } );
+        advance_time( 11 );
+        my $testid2 = $rpcapi->start_domain_test( { domain => "zone1.rpcapi.example" } );
+
+        $rpcapi->{db}->test_progress( $testid1, 1 );    # mark test as started
```

Comment:
Have you considered using `get_test_request`?

Reply:
I'd forgotten about that! Thanks for pointing it out, I'll update it.

Reply:
I had another look at this and realized `test_progress` has an important advantage over `get_test_request` - it lets you specify a test id. This makes the test robust against unrelated tests in the database waiting to be allocated.